.github/workflows/ci.yml

```diff
@@ -38,28 +38,6 @@ jobs:
     - uses: actions/checkout@v5
       with:
         persist-credentials: false
-    # update the version
```

Comment:
If you remove these, you get something like:

```
Name: rdmo
Version: 0.1.dev1+ge9e0fd81f
```

in the built wheel in a PR, is this desired?

Ref: https://github.com/rdmorganiser/rdmo/actions/runs/19539625158#summary-55942209390

Reply:
Nice catch. Actually I saw that one before with the fetch-depth: 0. Did not remember.

Reply:
Yes, this worked. `version_scheme = "release-branch-semver"` uses `2.4.0` unless the branch name looks like `2.3.x`, which is what we want, right?